microapps/FeatureFormsApp/app/src/main/java/com/arcgismaps/toolkit/featureformsapp/screens/login/LoginViewModel.kt

```diff
@@ -22,23 +22,40 @@ import com.arcgismaps.portal.Portal
 import com.arcgismaps.toolkit.authentication.AuthenticatorState
 import com.arcgismaps.toolkit.featureformsapp.BuildConfig
 import com.arcgismaps.toolkit.featureformsapp.data.PortalSettings
+import com.arcgismaps.toolkit.featureformsapp.data.local.UrlEntry
+import com.arcgismaps.toolkit.featureformsapp.data.local.UrlHistoryDao
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withTimeoutOrNull
 import javax.inject.Inject
 
 @HiltViewModel
 class LoginViewModel @Inject constructor(
-    private val portalSettings: PortalSettings
+    private val portalSettings: PortalSettings,
+    private val urlHistoryDao: UrlHistoryDao
 ) : ViewModel() {
 
     private data class Credentials(val username: String = "", val password: String = "")
 
     val authenticatorState = AuthenticatorState()
 
+    val urlHistory: StateFlow<List<String>> = combine(urlHistoryDao.observeAll()) { res ->
+        res.first().map {
+            it.url
+        }
+    }.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(1000),
+        initialValue = emptyList()
+    )
+
```

Comment:
this doesn't seem like the intended use of `combine` because it doesn't combine two flows. Is this more concise?
```suggestion
    val urlHistory: StateFlow<List<String>> = urlHistoryDao.observeAll().transform<List<UrlEntry>, List<String>> { res ->
        res.map {
            it.url
        }
    }.stateIn(
        scope = viewModelScope,
        started = SharingStarted.WhileSubscribed(1000),
        initialValue = emptyList()
    )

```

Reply:
I would even say to use `map` instead of `transform` because the types can be inferred, but having one suspending `map` call surrounding a non-suspending `map` call may be a bit of a cognitive load for the reader.

Reply:
I think using `map` makes sense.